samtranslator/validator/value_validator.py

```diff
@@ -86,11 +86,11 @@ def to_be_a_list_of(self, expected_type: ExpectedType, message: Optional[str] =
             ).to_be_a(expected_type, message)
         return value
 
-    def to_be_a_string(self, message: Optional[str] = "") -> T:
-        return self.to_be_a(ExpectedType.STRING, message)
+    def to_be_a_string(self, message: Optional[str] = "") -> str:
+        return cast(str, self.to_be_a(ExpectedType.STRING, message))
 
-    def to_be_an_integer(self, message: Optional[str] = "") -> T:
-        return self.to_be_a(ExpectedType.INTEGER, message)
+    def to_be_an_integer(self, message: Optional[str] = "") -> str:
+        return cast(str, self.to_be_a(ExpectedType.INTEGER, message))
```

Comment:
I'm confused by the name and return type

Reply:
this method raise exception if the value to validate is not an integer, and return the value "as an integer" to make mypy know it is now an integer.

(I immediately cannot read this sentence after I finished it lol)

Reply:
@acristin added more docsrtring

Reply:
>  and return the value "as an integer"

That's where my confusion comes from; the method returns `str`

Reply:
yeah, it either return a `str` (if the value is a string, happy case), or raise excepion

Reply:
fixed... I see where is the confusion